src/fortranscanner.l

```diff
@@ -1826,13 +1826,17 @@ const char* prepassFixedForm(const char* contents, int *hasContLine,int fixedCom
     free(newContents);
     return NULL;
   }
-  newContentsSize = (int)strlen(newContents);
-  if (newContents[newContentsSize - 1] != '\n')
+
+  if (newContents[j] == '\n')
```

Comment:
Looks like if `strlen(contents)==0` then `j==-1` so this would access the array out of bounds...

Reply:
Might happen but is highly unlikely, created a protection for this case.